isb_lib/utilities/requests_utilities.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import random
+
+import requests
+from requests import RequestException, Timeout, Response
+
+L = logging.getLogger("requests_utilities")
+
+
+class RetryingRequests:
+    def __init__(self, include_random_on_failure: bool = False, timeout=60, max_retries=10):
+        self._include_random_on_failure = include_random_on_failure
+        self._timeout = timeout
+        self._max_retries = max_retries
+
+    def get(self, url: str, params: dict = None, headers: dict = None) -> Response:
```

Comment:
Do you want to include a pause between retries? Or is that done elsewhere?

Reply:
It is not.  That is a good suggestion!